rb_ws/src/buggy/buggy/serial/ros_to_bnyahaj.py

```diff
@@ -0,0 +1,255 @@
+            Bool, self_name + "/buggy/debug/use_auton_steer", 1
+        )
+        self.rc_uplink_qual_publisher = self.create_publisher(
+            UInt8, self_name + "/buggy/debug/rc_uplink_quality", 1
+        )
+        self.nand_fix_publisher = self.create_publisher(
+            UInt8, self_name + "/buggy/debug/nand_fix", 1
+        )
+
+    def set_alarm(self, msg):
+        """
+        alarm ros topic reader, locked so that only one of the setters runs at once
+        """
+        with self.lock:
+            self.get_logger().debug(f"Reading alarm of {msg.data}")
+            self.alarm = msg.data
+
+    def set_steering(self, msg):
+        """
+        Steering Angle Updater, updates the steering angle locally if updated on ros stopic
+        """
+        self.get_logger().debug(f"Read steeering angle of: {msg.data}")
+        # print("Steering angle: " + str(msg.data))
+        # print("SET STEERING: " + str(msg.data))
+        with self.lock:
+            self.steer_angle = msg.data
+            self.fresh_steer = True
+
+    def writer_thread(self):
+        """
+        Sends ROS Topics to bnayhaj serial, only sends a steering angle when we receive a fresh one
+        Will send steering and alarm node.
+        TODO: Does alarm node exist for NAND?
+        """
+        self.get_logger().info("Starting sending alarm and steering to teensy!")
+
+        while not self.is_shutdown():
+            if self.fresh_steer:
+                with self.lock:
+                    self.comms.send_steering(self.steer_angle)
+                    self.fresh_steer = False
+
+            with self.lock:
+                self.comms.send_alarm(self.alarm)
+
+            self.steer_send_rate.sleep()
+
+    def reader_thread(self):
+        """
+        Reads three different types of packets, that should have better ways of differntiating
+        Odometry -> (SC) NAND Odomotery
+        BnyaTelemetry -> (NAND) Self Odom
+        tuple -> (SC, maybe NAND?) Debug Info
+        """
+        self.get_logger().info("Starting reading odom from teensy!")
+        while rclpy.ok():
+            packet = self.comms.read_packet()
+
+            if isinstance(packet, Odometry):
+                self.get_logger().debug("packet" + str(packet))
+                # Publish to odom topic x and y coord
+                odom = ROSOdom()
+                # convert to long lat
+                try:
+                    lat, long = World.utm_to_gps(packet.y, packet.x)
+                    odom.pose.pose.position.x = long
+                    odom.pose.pose.position.y = lat
+                    self.odom_publisher.publish(odom)
+                except Exception as e:
+                    self.get_logger().warn(
+                        "Unable to convert other buggy position to lon lat" + str(e)
+                    )
+
+            elif isinstance(packet, BnyaTelemetry):
+                self.get_logger().debug("packet" + str(packet))
```

Comment:
Rather than having a seperate elif on each if case, have a standard debug print for str(packet)

Reply:
Moved the debug for each packet outside of the if/elif blocks. Note that this means tuple packets will also be logged now.